src/hooks/useVaaVerifier.ts

```diff
@@ -0,0 +1,128 @@
+import {
+  ChainId,
+  ParsedVaa,
+  hexToUint8Array,
+  parseNFTPayload,
+  parseTransferPayload,
+  parseVaa,
+  tryHexToNativeString,
+} from "@certusone/wormhole-sdk";
+import { useMemo } from "react";
+import {
+  getNFTBridgeAddressForChain,
+  getTokenBridgeAddressForChain,
+} from "../utils/consts";
+
+enum VAAType {
+  UNKNOWN = 0,
+  TRANSFER = 1,
+  ATTEST = 2,
+  TRANSFER_WITH_PAYLOAD = 3,
+}
+
+interface ParserPayload<T> {
+  (buffer: Buffer): T;
+}
+
+function parsePayload<T>(
+  payload: Buffer,
+  parser: ParserPayload<T>
+): T | undefined {
+  try {
+    if (payload) {
+      return parser(payload);
+    }
+  } catch (err) {
+    console.debug(`Failed to parse payload with ${parser}`, err);
+  }
+}
+
+function getPayloadType(payload: Buffer): VAAType {
+  try {
+    const type = payload.readInt8(0);
+    if (type > 0 || type <= 3) {
+      return type;
+    }
+  } catch (err) {
+    console.debug(`Failed to parse type from payload`, err);
+  }
+  return VAAType.UNKNOWN;
+}
+
+function parseVaaFromHexString(hexVaa: string): ParsedVaa {
+  try {
+    return parseVaa(hexToUint8Array(hexVaa));
+  } catch (err) {
+    console.debug(err);
+  }
+  return {} as ParsedVaa;
+}
+
+function itWasEmittedFrom(
+  emitterAddress: Buffer,
+  emitterChain: number,
+  getAddressFn: (chainId: ChainId) => string
```

Comment:
Do you need to pass this function here? Its invocation can be done outside since it only depends on the `emitterChain`

Reply:
the intention is to use high-order functions to archive composability, so, in future updates, if we want to support another contract address than the token bridge or nft bridge, we need to provide a new function as an argument, also for clarity, making it code readable. But, let me take a look if I can improve it